### ISSUE TITLE
Error when reading datapackage with default media value

### ISSUE THREAD

**Issue by SanderDevisscher** [inbo/camtraptor#338]
When running [this code](https://github.com/inbo/fis-projecten/blob/49dc94b2a6a8ca71a3e3d7d92b32968f109938f5/Grofwild/Voer_Yse/scripts/Voer_Yse_verkenning_cameravallen.Rmd#L134) I get the error below:
`VY <- read_camtrap_dp("./Grofwild/Voer_Yse/Data/Input/Agouti/FilesVY")`

```
<error/purrr_error_indexed>
Error in `map()`:
ℹ In index: 1.
Caused by error:
! Invalid input type, expected 'list' actual 'NULL'
---
Backtrace:
     ▆
  1. ├─camtraptor::read_camtrap_dp("./Grofwild/Voer_Yse/Data/Input/Agouti/FilesVY/datapackage.json")
  2. │ └─frictionless::read_resource(package, "media") at camtraptor/R/read_camtrap_dp.R:129:5
  3. │   └─frictionless:::read_from_path(package, resource_name, col_select)
  4. │     └─purrr::map_df(...)
  5. │       └─purrr::map(.x, .f, ...)
  6. │         └─purrr:::map_("list", .x, .f, ..., .progress = .progress)
  7. │           ├─purrr:::with_indexed_errors(...)
  8. │           │ └─base::withCallingHandlers(...)
  9. │           ├─purrr:::call_with_cleanup(...)
 10. │           └─frictionless (local) .f(.x[[i]], ...)
 11. │             └─readr::read_delim(...)
 12. │               └─vroom::vroom(...)
 13. │                 └─vroom:::vroom_(...)
 14. └─base::.handleSimpleError(...)
 15.   └─purrr (local) h(simpleError(msg, call))
 16.     └─cli::cli_abort(...)
 17.       └─rlang::abort(...)
```
Adding `media = FALSE` seems to fix the issue, however I do get the following message:
```
The dataset uses Camtrap DP version 1.0, it has been converted to 0.1.6.                                                
See https://inbo.github.io/camtraptor/#camtrap-dp for details. 
```

```
sessionInfo()
R version 4.4.1 (2024-06-14 ucrt)
Platform: x86_64-w64-mingw32/x64
Running under: Windows 10 x64 (build 19045)

Matrix products: default


locale:
[1] LC_COLLATE=Dutch_Belgium.utf8  LC_CTYPE=Dutch_Belgium.utf8    LC_MONETARY=Dutch_Belgium.utf8
[4] LC_NUMERIC=C                   LC_TIME=Dutch_Belgium.utf8    

time zone: Europe/Brussels
tzcode source: internal

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
 [1] fistools_1.2.7    camtraptor_0.25.0 unmarked_1.4.3    viridis_0.6.5     viridisLite_0.4.2 reshape2_1.4.4   
 [7] ggplotify_0.1.2   overlap_0.3.9     suntools_1.0.0    activity_1.3.4    here_1.0.1        googledrive_2.1.1
[13] mapview_2.11.2    lubridate_1.9.3   forcats_1.0.0     stringr_1.5.1     dplyr_1.1.4       purrr_1.0.2      
[19] readr_2.1.5       tidyr_1.3.1       tibble_3.2.1      tidyverse_2.0.0   INBOtheme_0.6.0   raster_3.6-26    
[25] leaflet_2.2.2     rgdal_1.6-7       sp_2.1-4          ggplot2_3.5.1     knitr_1.48       

loaded via a namespace (and not attached):
 [1] rstudioapi_0.16.0       jsonlite_1.8.8          magrittr_2.0.3          fs_1.6.4               
 [5] vctrs_0.6.5             memoise_2.0.1           askpass_1.2.0           base64enc_0.1-3        
 [9] terra_1.7-78            htmltools_0.5.8.1       usethis_3.0.0           curl_5.2.1             
[13] gridGraphics_0.5-1      svDialogs_1.1.0         KernSmooth_2.23-24      htmlwidgets_1.6.4      
[17] desc_1.4.3              plyr_1.8.9              cachem_1.1.0            conflicted_1.2.0       
[21] mime_0.12               lifecycle_1.0.4         pkgconfig_2.0.3         R6_2.5.1               
[25] fastmap_1.2.0           shiny_1.9.1             digest_0.6.36           colorspace_2.1-1       
[29] ps_1.7.7                rprojroot_2.0.4         leafem_0.2.3            pkgload_1.4.0          
[33] crosstalk_1.2.1         fansi_1.0.6             timechange_0.3.0        httr_1.4.7             
[37] compiler_4.4.1          gargle_1.5.2            proxy_0.4-27            remotes_2.5.0          
[41] bit64_4.0.5             withr_3.0.1             DBI_1.2.3               pkgbuild_1.4.4         
[45] aws.ec2metadata_0.2.0   MASS_7.3-61             openssl_2.2.1           rappdirs_0.3.3         
[49] sessioninfo_1.2.2       classInt_0.4-10         tools_4.4.1             units_0.8-5            
[53] httpuv_1.6.15           glue_1.7.0              satellite_1.0.5         callr_3.7.6            
[57] promises_1.3.0          grid_4.4.1              sf_1.0-16               generics_0.1.3         
[61] gtable_0.3.5            tzdb_0.4.0              class_7.3-22            hms_1.1.3              
[65] utf8_1.2.4              pillar_1.9.0            yulab.utils_0.1.7       vroom_1.6.5            
[69] later_1.3.2             frictionless_1.2.0.9000 lattice_0.22-6          bit_4.0.5              
[73] tidyselect_1.2.1        miniUI_0.1.1.1          gridExtra_2.3           stats4_4.4.1           
[77] xfun_0.46               devtools_2.4.5          stringi_1.8.4           codetools_0.2-20       
[81] cli_3.6.3               xtable_1.8-4            munsell_0.5.1           processx_3.8.4         
[85] svGUI_1.0.1             Rcpp_1.0.13             png_0.1-8               parallel_4.4.1         
[89] ellipsis_0.3.2          assertthat_0.2.1        profvis_0.3.8           urlchecker_1.0.1       
[93] scales_1.3.0            fortunes_1.5-4          e1071_1.7-14            crayon_1.5.3           
[97] rlang_1.1.4           
```

**Comment by damianooldoni** [comment]
Hi @SanderDevisscher. Everybody is having this error, see #334. Agouti devs added a non standard column without changing the fields in the schema passed by the json file. It will be adjusted soon at Agouti level soon.

**Comment by damianooldoni** [comment]
@SanderDevisscher: if you are not using media, maybe you can use `read_camtrap_dp(path_to_your_json_file, media = FALSE)`. Same tip as I gave in #334.

**Comment by SanderDevisscher** [comment]
@damianooldoni the script I'm reviewing does use the media at least I thinks so. I found the adding `media = FALSE` solution as well but this causes the following error to occur `can't find media in .$data` when trying to run `get_record_table()`.

**Comment by SanderDevisscher** [comment]
@damianooldoni I see this is actually a duplicate of #334

**Comment by SanderDevisscher** [comment]
@damianooldoni I think it is something different when @MartijnUH runs the same code on the same data he doesn't get the same issue. He gets this:
```
Warning: The `path` argument of `read_camtrap_dp()` is deprecated as of camtraptor 0.6.0.
ℹ Use argument `file` containing the path or URL to the `datapackage.json` file. The use of
  parameter `path` with path to the local directory is deprecated since version 0.6.0.
This warning is displayed once every 8 hours.
Call `lifecycle::last_lifecycle_warnings()` to see where this warning was generated.
Warning: One or more parsing issues, call `problems()` on your data frame for details, e.g.:
  dat <- vroom(...)
  problems(dat)
Warning in check_reading_issues(media_df, "media") :
  One or more parsing issues occurred while reading `media`. Check `?read_camtrap_dp()` for examples on how to use `readr::problems()`.
The dataset uses Camtrap DP version 1.0, it has been converted to 0.1.6.
See https://inbo.github.io/camtraptor/#camtrap-dp for details.
``` 
Here is his sessioninfo:

```
R version 4.4.0 (2024-04-24 ucrt)
Platform: x86_64-w64-mingw32/x64
Running under: Windows 10 x64 (build 19045)

Matrix products: default


locale:
[1] LC_COLLATE=English_United Kingdom.utf8  LC_CTYPE=English_United Kingdom.utf8   
[3] LC_MONETARY=English_United Kingdom.utf8 LC_NUMERIC=C                           
[5] LC_TIME=English_United Kingdom.utf8    

time zone: Europe/Brussels
tzcode source: internal

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
 [1] fistools_1.0.0       camtraptor_0.25.0    unmarked_1.4.1       viridis_0.6.5       
 [5] viridisLite_0.4.2    reshape2_1.4.4       ggplotify_0.1.2      overlap_0.3.9       
 [9] suntools_1.0.0       activity_1.3.4       here_1.0.1           googledrive_2.1.1   
[13] mapview_2.11.2       lubridate_1.9.3.9000 forcats_1.0.0        stringr_1.5.1       
[17] dplyr_1.1.4          purrr_1.0.2          readr_2.1.5          tidyr_1.3.1         
[21] tibble_3.2.1         tidyverse_2.0.0      INBOtheme_0.6.0      raster_3.6-26       
[25] leaflet_2.2.2        rgdal_1.5-32         sp_2.1-4             ggplot2_3.5.1       
[29] knitr_1.47          

loaded via a namespace (and not attached):
 [1] DBI_1.2.3               gridExtra_2.3           remotes_2.5.0           conflicted_1.2.0       
 [5] rlang_1.1.3             magrittr_2.0.3          e1071_1.7-14            compiler_4.4.0         
 [9] png_0.1-8               vctrs_0.6.5             profvis_0.3.8           crayon_1.5.3           
[13] pkgconfig_2.0.3         fastmap_1.2.0           ellipsis_0.3.2          leafem_0.2.3           
[17] utf8_1.2.4              promises_1.3.0          rmarkdown_2.27          sessioninfo_1.2.2      
[21] tzdb_0.4.0              bit_4.0.5               xfun_0.44               satellite_1.0.5        
[25] cachem_1.1.0            jsonlite_1.8.8          later_1.3.2             svDialogs_1.1.0        
[29] terra_1.7-78            parallel_4.4.0          R6_2.5.1                stringi_1.8.4          
[33] pkgload_1.3.4           Rcpp_1.0.12             assertthat_0.2.1        usethis_2.2.3          
[37] fortunes_1.5-4          base64enc_0.1-3         httpuv_1.6.15           timechange_0.3.0       
[41] tidyselect_1.2.1        rstudioapi_0.16.0       yaml_2.3.8              codetools_0.2-20       
[45] miniUI_0.1.1.1          curl_5.2.1              pkgbuild_1.4.4          lattice_0.22-6         
[49] plyr_1.8.9              shiny_1.8.1.1           withr_3.0.0             askpass_1.2.0          
[53] evaluate_0.24.0         gridGraphics_0.5-1      sf_1.0-16               units_0.8-5            
[57] proxy_0.4-27            urlchecker_1.0.1        pillar_1.9.0            frictionless_1.1.0.9000
[61] KernSmooth_2.23-24      stats4_4.4.0            generics_0.1.3          vroom_1.6.5            
[65] rprojroot_2.0.4         hms_1.1.3               munsell_0.5.1           scales_1.3.0           
[69] svGUI_1.0.1             xtable_1.8-4            class_7.3-22            glue_1.7.0             
[73] tools_4.4.0             fs_1.6.4                grid_4.4.0              crosstalk_1.2.1        
[77] devtools_2.4.5          colorspace_2.1-0        cli_3.6.2               rappdirs_0.3.3         
[81] fansi_1.0.6             gargle_1.5.2            gtable_0.3.5            yulab.utils_0.1.4      
[85] digest_0.6.35           classInt_0.4-10         htmlwidgets_1.6.4       memoise_2.0.1          
[89] htmltools_0.5.8.1       lifecycle_1.0.4         httr_1.4.7              mime_0.12              
[93] bit64_4.0.5             openssl_2.2.0           MASS_7.3-60.2
```

**Comment by SanderDevisscher** [comment]
@damianooldoni thanks for the insights. Since the issue will be fixed on agouti's side as stated in #334. I'll ignore the error for now.
